server/src/main/java/org/opensearch/cluster/service/MasterService.java

```diff
@@ -299,8 +299,15 @@ public static boolean assertNotMasterUpdateThread(String reason) {
     }
 
     private void runTasks(TaskInputs taskInputs) {
-        final String longSummary = logger.isTraceEnabled() ? taskInputs.taskSummaryGenerator.apply(true) : "";
-        final String shortSummary = taskInputs.taskSummaryGenerator.apply(false);
+        final String longSummary;
+        final String shortSummary;
+        if (taskInputs.updateTasks.size() <= 1000) {
+            longSummary = taskInputs.taskSummaryGenerator.apply(true);
+            shortSummary = longSummary;
+        } else {
+            longSummary = logger.isTraceEnabled() ? taskInputs.taskSummaryGenerator.apply(true) : "";
+            shortSummary = taskInputs.taskSummaryGenerator.apply(false);
```

Comment:
it will compute twice now. 

shouldnt this be:

```
if(logger.isTraceEnabled()) {
    summary = taskInputs.taskSummaryGenerator.apply(true);
} else {
    summary = taskInputs.taskSummaryGenerator.apply(false);
}
```

Now, for shortSummary, it already takes care of logging n tasks (which is 1000 here). We can remove multiple if check on tracelog and simply use summary everywhere?

Reply:
We were logging INFO at few places with short summary irrespective of trace log level and only with long summary where trace log was required.

For now refactoring as suggested to keep code simple.

Reply:
what was the being logged before from those info logs before you made long/short summary change?

Reply:
long summary only.